tests/functional/pv/pvc_clone/test_clone_deletion_after_max_cluster_space_utilization.py

```diff
@@ -0,0 +1,197 @@
+from ocs_ci.utility.utils import TimeoutSampler
+from ocs_ci.ocs.cluster import change_ceph_full_ratio, CephCluster
+from ocs_ci.helpers import helpers
+from ocs_ci.ocs.exceptions import TimeoutExpiredError
+from ocs_ci.utility.prometheus import PrometheusAPI
+from ocs_ci.framework.pytest_customization.marks import (
+    skipif_external_mode,
+    magenta_squad,
+)
+
+logger = logging.getLogger(__name__)
+
+
+@tier2
+@pytest.mark.parametrize(
+    argnames=["interface_type"],
+    argvalues=[
+        pytest.param(constants.CEPHBLOCKPOOL),
+        pytest.param(constants.CEPHFILESYSTEM),
+    ],
+)
+class TestCloneDeletion(E2ETest):
+    """
+    Tests to verify clone deletion without error
+    after cluster out of full ratio
+    """
+
+    @pytest.fixture(autouse=True)
+    def setup(self, interface_type, pvc_factory, pod_factory, request):
+        """
+        create resources for the test
+
+        Args:
+        interface_type(str): The type of the interface
+            (e.g. CephBlockPool, CephFileSystem)
+        pvc_factory: A fixture to create new pvc
+        pod_factory: A fixture to create new pod
+
+        """
+
+        def teardown():
+            # change ceph full ratio to standard value
+            change_ceph_full_ratio(85)
+
+        request.addfinalizer(teardown)
+
+        logger.info("Starting the test setup")
+        # Getting the total Storage capacity
+        self.ceph_cluster = CephCluster()
+        self.ceph_capacity = int(self.ceph_cluster.get_ceph_capacity())
+        logger.info(f"ceph_capacity: {self.ceph_capacity}")
+
+        # Getting the free Storage capacity
+        self.ceph_free_capacity = int(self.ceph_cluster.get_ceph_free_capacity())
+        logger.info(f"ceph_free_capacity: {self.ceph_free_capacity}")
+
+        # Change ceph full ratio
+        change_ceph_full_ratio(10)
+
+        # Use 10% of the free storage capacity in the test
+        self.capacity_to_use = int(self.ceph_free_capacity * 0.10)
+        logger.info(f"capacity_to_use: {self.capacity_to_use}")
+
+        self.num_of_clones = 10
+        # Calculating the PVC size in GiB
+        self.pvc_size = int(self.capacity_to_use / (self.num_of_clones + 1))
+        logger.info(f"pvc size: {self.pvc_size}")
+
+        self.pvc_obj = pvc_factory(
+            interface=interface_type, size=self.pvc_size, status=constants.STATUS_BOUND
+        )
+        self.pod_obj = pod_factory(
+            interface=interface_type, pvc=self.pvc_obj, status=constants.STATUS_RUNNING
+        )
+
+        # Calculating the file size as 95% of the PVC size - in MB
+        self.filesize = f"{int(self.pvc_size * 1024 * 0.95)}M"
+
+        self.pod_obj.run_io(
+            size=self.filesize,
+            io_direction="write",
+            storage_type="fs",
+        )
+
+        self.pod_obj.get_fio_results()
+        logger.info(f"IO finished on pod {self.pod_obj.name}")
+
+    def verify_alerts_via_prometheus(self, expected_alerts, threading_lock):
```

Comment:
This func also exists in test_cluster_full_and_recovery.py , please move this to some common place and call it in the test case

Reply:
done